packages/dev/core/src/Meshes/Compression/Draco/dracoCompression.encoder.ts

```diff
@@ -0,0 +1,338 @@
+import type { IGetSetVerticesData } from "../../mesh.vertexData";
+import { VertexBuffer } from "../../../Buffers/buffer";
+import type { Nullable, FloatArray } from "../../../types";
+import { DracoAttributeName, DracoAttribute, DracoGeometryType, DracoEncoderMethod, DracoCompressionBase } from "./dracoCommons";
+import type { IDracoEncoder, IDracoCompressionEngineConfiguration, IDracoEncodedPrimitive, IDracoEncoderOptions } from "./dracoCommons";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+declare let DracoEncoderModule: any;
+declare function importScripts(...urls: string[]): void;
+declare function postMessage(message: any, transfer?: any[]): void;
+
+/////////////////////////////////
+//    BEGIN WORKER CONTEXT     //
+/////////////////////////////////
+
+/**
+ * The encoder function that gets converted to a blob url to pass into a worker.
+ * @param encoderModule
+ * @param attributes
+ * @param indices
+ * @param geometryDestination the type of geometry. TRIANGULAR_MESH = 0, POINT_CLOUD = 1
+ * @param options
+ * @param destination
+ */
+function encodeMesh(
+    encoderModule: any,
+    attributes: Array<{ key: string; type: number; stride: number; data: Float32Array }>,
+    indices: Nullable<Uint16Array | Uint32Array>,
+    geometryDestination: 0 | 1,
+    options: any,
+    destination?: ArrayBuffer
+): Nullable<{ data: Uint8Array; attributes: { [kind: string]: number } }> {
+    const vertices = attributes.find((a) => a.key == "position")?.data;
+
+    if (vertices?.length) {

```

Comment:
Should this be a failure? Returning null to the user when there are no vertices is more complicated since the contract now needs to be nullable. Whoever is calling `encodeMeshAsync` on the encoder can easily check to see if the mesh being encoded has vertices.